### DISCUSSION TITLE
Is there any way to delete only 1 specific widget in a page?

### DISCUSSION THREAD

**Discussion by ag-gaphp** [py-pdf/pypdf#2662] [Q&A] [answered]
From the docs: https://pypdf.readthedocs.io/en/stable/modules/PdfWriter.html#pypdf.PdfWriter.remove_annotations

It seems that this method will delete _ALL_ annotations of the provided type from a page. Is it possible to only delete 1 widget on a page using pypdf?

**Comment by stefan6419846** [comment]
As far as I am aware, there is no builtin method for this. As a workaround, you might want to have a look at `_remove_annots_from_page` and tweak this to your needs to only delete the exact annotation you are looking for.

**Reply by ag-gaphp** [reply]
I had the same idea, but still trying to wrap my head around all the pieces to tweak it to my needs. I was hoping there was already a way to do it since that method existed. I'll keep plugging away at it, thanks for the suggestion.

**Reply by pubpub-zz** [reply, marked as answer]
For me you should be able to loop through annotations and then delete it:
(should be something like this)
```
import pypdf
w = pypdf.PdfWriter(clone_from="input.pdf")
to_be_deleted = []
for i, a in enumerate(w.pages[0]["/Annots"]):
   a=a.get_object()
   if #### criteria on a ####:
       to_be_deleted.append(i)
for i in to_be_deleted[::-1]:
   del w.pages[0]["/Annots"][i]

w.write("output.pdf")
```

**Reply by ag-gaphp** [reply]
Much appreciated @pubpub-zz ! This was perfect. I had the right idea, but my execution attempts were way, way off. This is pretty simple and helped me understand a little more about how pypdf does things.
